app/src/main/java/zapsolutions/zap/HomeActivity.java

```diff
@@ -112,6 +114,8 @@ protected void onCreate(Bundle savedInstanceState) {
         mInputMethodManager = (InputMethodManager) this.getSystemService(Context.INPUT_METHOD_SERVICE);
         mHandler = new Handler();
 
+        mUnlockDialog = buildUnlockDialog();
```

Comment:
We only create the dialog once and hold a reference, so we can later check if it is already shown.